src/run.rs

```diff
@@ -187,7 +186,12 @@ pub fn run() {
     .collect::<Vec<&str>>();
 
   let justfile_option = matches.value_of("JUSTFILE");
-  let working_directory_option = matches.value_of("WORKING-DIRECTORY");
+  let mut working_directory_option = matches.value_of("WORKING-DIRECTORY");
+
+  if justfile_option.is_some() && working_directory_option.is_none() {
+    let justfile_path = Path::new(justfile_option.unwrap()).parent();
```

Comment:
If `parent()` returns none, we should exit with a message, like `die!("Could not find parent directory of justfile at `{}`", justfile)`

Reply:
Added none check. But there seems to be a known issue(https://github.com/rust-lang/rust/issues/36861) with parent returning "" in certain cases. Please suggest if it can be handled in a better way.

Reply:
Thanks for the changes! Let's resolve the justfile to an absolute path, and then using the parent of that.

Reply:
That should fix the issue with parent returning `""`

Reply:
Disclaimer: I am very new to Rust :-), and although I understand reference concept, not able to find a 'Rust way' to deal with following situation and so having trouble with the code: 

```
if let (Some(justfile), None) = (justfile_option, working_directory_option) {
     let justfile_path = Path::new(justfile);
    if Path::new(justfile).is_absolute() {
        working_directory_option = Path::new(justfile).parent().unwrap().to_str();
    } else {
      let justfile_path_canonical = justfile_path.canonicalize();
      if justfile_path.canonicalize().is_ok() {
// @casey following line errors out with borrowed value does not live long enough. Any suggestion?
        working_directory_option = justfile_path_canonical.ok().unwrap().to_str(); 
      } else {
        die!("Could not find parent directory of justfile at {}.", justfile);
      }
    }
    //println!("working_directory_option:{:?}", working_directory_option);
  }

```

Reply:
No worries at all, we were all new to Rust once :)

Can you push the code (even if it's broken) to the PR branch so I can try it out?

Reply:
Done. It's broken.